docs/apworld specification.md

```diff
@@ -23,18 +23,61 @@ Metadata about the apworld is defined in an `archipelago.json` file inside the z
 The current format version has at minimum:
 ```json
 {
-    "version": 6,
-    "compatible_version": 5,
+    "version": 7,
+    "compatible_version": 7,
     "game": "Game Name"
 }
 ```
```

Comment:
I have a strong feeling we're going to see a lot of people manually adding an archipelago.json with just these 3 fields filled out, from people who just skim the docs. Not sure how much that actually matters, just pointing out the potential footgun